### ISSUE TITLE
Conversion stops when attachment cannot be found

### ISSUE THREAD

**Issue by yuergen** [qqilihq/partial-emlx-converter#4]
It seems that Apple Mail loses a lot of attachments, when mail has been stored for years. When I try to convert my mailboxes, I get lots of errors like these:
```
Processing 728F079D-C331-494A-844E-885BC16E5661/_Neu.mbox/E401EF12-8C7D-40D6-B4A7-784D2410BE37/Data/8/8/Messages/88886.partial.emlx
(node:4391) UnhandledPromiseRejectionWarning: Error: ENOENT: no such file or directory, open '/home/juergen/mail/V5/728F079D-C331-494A-844E-885BC16E5661/_Neu.mbox/E401EF12-8C7D-40D6-B4A7-784D2410BE37/Data/8/8/Attachments/88886/1/Technical English.zip'
    at Object.fs.openSync (fs.js:646:18)
    at Object.fs.readFileSync (fs.js:551:33)
    at transformRec (/home/juergen/github/partial-emlx-converter/build/src/converter.js:116:29)
    at parts.forEach (/home/juergen/github/partial-emlx-converter/build/src/converter.js:98:36)
    at Array.forEach (<anonymous>)
    at transform (/home/juergen/github/partial-emlx-converter/build/src/converter.js:98:11)
    at emlformat.parse (/home/juergen/github/partial-emlx-converter/build/src/converter.js:49:17)
    at Object.emlformat.parse (/home/juergen/github/partial-emlx-converter/node_modules/eml-format/lib/eml-format.js:554:5)
    at Promise (/home/juergen/github/partial-emlx-converter/build/src/converter.js:45:19)
    at new Promise (<anonymous>)
(node:4391) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 3)
(node:4391) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

I suggest skipping these mails during conversion and maybe storing them in a folder named rejects. I am doing that manually now.

I have no idea about node.js, so I will not be able to provide code for this. :(

Aside from that partial-emlx-converter is a great tool. Thank you for your effort!

**Comment by yuergen** [comment]
Mail.app seems to store some attachments as directories. If that happens the following output is given:
```
Processing 728F079D-C331-494A-844E-885BC16E5661/[Google Mail].mbox/Gesendet.mbox/E401EF12-8C7D-40D6-B4A7-784D2410BE37/Data/4/4/1/Messages/144385.partial.emlx
(node:6501) UnhandledPromiseRejectionWarning: Error: EISDIR: illegal operation on a directory, read
    at Object.fs.readSync (fs.js:675:18)
    at tryReadSync (fs.js:540:20)
    at Object.fs.readFileSync (fs.js:583:19)
    at transformRec (/home/juergen/github/partial-emlx-converter/build/src/converter.js:116:29)
    at parts.forEach (/home/juergen/github/partial-emlx-converter/build/src/converter.js:98:36)
    at Array.forEach (<anonymous>)
    at transform (/home/juergen/github/partial-emlx-converter/build/src/converter.js:98:11)
    at emlformat.parse (/home/juergen/github/partial-emlx-converter/build/src/converter.js:49:17)
    at Object.emlformat.parse (/home/juergen/github/partial-emlx-converter/node_modules/eml-format/lib/eml-format.js:554:5)
    at Promise (/home/juergen/github/partial-emlx-converter/build/src/converter.js:45:19)
(node:6501) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 3)
(node:6501) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

**Comment by qqilihq** [comment]
Thanks for the feedback, I already started working on a fix for this issue, based on the feedback in #3 -- I'll keep you posted!

**Comment by yuergen** [comment]
Thank you! :)

**Comment by yuergen** [comment]
I just saw you closed issue #3 and tested the changed code on my mail archive. I have mails, where there is a name given, but no extension:
```
--------------040005040409050603090005
Content-Transfer-Encoding: BASE64
Content-Disposition: ATTACHMENT;
        filename=Warnmeldung_unbekannter_Art
Content-Type: IMAGE/PNG;
        name=Warnmeldung_unbekannter_Art
X-Apple-Content-Length: 138412
```

This gives this error:
```
Processing F110B03C-8868-4AE1-8DCA-D123205864A3/INBOX.mbox/2014.mbox/E401EF12-8C7D-40D6-B4A7-784D2410BE37/Data/9/2/2/Messages/229417.partial.emlx
(node:24147) UnhandledPromiseRejectionWarning: Error: ENOENT: no such file or directory, open '/home/juergen/mail/V5/F110B03C-8868-4AE1-8DCA-D123205864A3/INBOX.mbox/2014.mbox/E401EF12-8C7D-40D6-B4A7-784D2410BE37/Data/9/2/2/Attachments/229417/2/Warnmeldung_unbekannter_Art'
```

The file exists with an extension in the name:
```
juergen@ws00 ~/m/V5> ls -la /home/juergen/mail/V5/F110B03C-8868-4AE1-8DCA-D123205864A3/INBOX.mbox/2014.mbox/E401EF12-8C7D-40D6-B4A7-784D2410BE37/Data/9/2/2/Attachments/229417/2/Warnmeldung_unbekannter_Art*
-rw-r--r-- 1 juergen juergen 102385 Mai 11  2018 /home/juergen/mail/V5/F110B03C-8868-4AE1-8DCA-D123205864A3/INBOX.mbox/2014.mbox/E401EF12-8C7D-40D6-B4A7-784D2410BE37/Data/9/2/2/Attachments/229417/2/Warnmeldung_unbekannter_Art.png
```

What were they smoking? ;-)

Can I submit test mails via adding them to the tests folder and then sending a pull request?

Thank you!

**Comment by qqilihq** [comment]
Wow, this becoming really a rabbit hole :-) I'll try to address further issues you posted, but didn't yet get to it.

> Can I submit test mails via adding them to the tests folder and then sending a pull request?

Yes, that would be awesome!

**Comment by qqilihq** [comment]
@yuergen I'll have some spare time during the weekend to look at the open bugs and issues. Wanted to double check: Are your test cases in the forked repo? Did you encounter any more problems?

**Comment by yuergen** [comment]
@qqilihq yes, I added two test cases, for which I seem to get similar errors. I will find some time to add some more. I suggest I add more test cases first, before I submit a pull request. Meanwhile you can download the tests from my repo. Is that okay for you?

**Comment by yuergen** [comment]
I hope a lot of the other partial emlx files can be converted once the test specimen can be converted. If failures remain, I will open a new pull request.

Mail.app stores mail in crazy ways. Especially in older versions. It really is a rabbit hole.

**Comment by qqilihq** [comment]
Thanks for test files -- in fact all test cases should be addressed now.

I've just released version 1.3 -- it's now available on NPM which should make things much more convenient. See here for details:

https://github.com/qqilihq/partial-emlx-converter#installation-with-package-manager

Let me know how it works!

**Comment by qqilihq** [comment]
I’ll close this as I haven’t heard back. Feel free to open a new ticket if you encounter any further issues.
